bundler/lib/dependabot/bundler/package_manager.rb

```diff
@@ -44,9 +43,14 @@ def initialize(version)
       def deprecated?
         deprecated_versions.include?(version)
       end
+
       sig { override.returns(T::Boolean) }
       def unsupported?
-        !deprecated? && version < supported_versions.first
+        # Check if the feature flag for Bundler v1 unsupported error is enabled.
+        return false unless Dependabot::Experiments.enabled?(:bundler_v1_unsupported_error)
+
+        # Determine if the Bundler version is unsupported.
+        version < supported_versions.first
```

Comment:
We expect the version to be lower then the first element of supported version if it is not supported. In other cases we are unable to determine if it is not supported.